src/core/Akka/IO/UdpConnection.cs

```diff
@@ -73,15 +73,17 @@ private void DoConnect(EndPoint address)
         {
             ReportConnectFailure(() =>
             {
-                _socket = new Socket(SocketType.Dgram, ProtocolType.Udp) { Blocking = false };
+                _socket = (_connect.Options.OfType<Inet.DatagramChannelCreator>().FirstOrDefault() ??
+                           new Inet.DatagramChannelCreator()).Create(address.AddressFamily);
+                _socket.Blocking = false;
 
                 foreach (var option in _connect.Options)
                 {
                     option.BeforeDatagramBind(_socket);
                 }
 
                 if (_connect.LocalAddress != null)
-                    _socket.Bind(_connect.LocalAddress);
```

Comment:
Should not use the local address provided in the `Connect` message, because it could be changed by DNS lookup